airflow/www/static/js/tree/Tree.jsx

```diff
@@ -0,0 +1,202 @@
+ * under the License.
+ */
+
+/* global localStorage */
+
+import React, { useRef, useState } from 'react';
+import {
+  Flex,
+  Box,
+  Text,
+  useDisclosure,
+  Collapse,
+  Switch,
+  FormControl,
+  FormLabel,
+  Spinner,
+} from '@chakra-ui/react';
+import { FiChevronUp, FiChevronDown } from 'react-icons/fi';
+
+import useTreeData from './useTreeData';
+import StatusBox from './StatusBox';
+import DagRunBar from './DagRunBar';
+import getMetaValue from '../meta_value';
+import { getDuration, formatDuration } from './utils';
+
+// dagId comes from dag.html
+const dagId = getMetaValue('dag_id');
+
+const DurationTick = ({ children, ...rest }) => (
+  <Text fontSize={10} color="gray.400" position="absolute" left="calc(100% + 6px)" whiteSpace="nowrap" {...rest}>
+    {children}
+  </Text>
+);
+
+const Tree = () => {
+  const containerRef = useRef();
+  const { data, refreshOn, onToggleRefresh } = useTreeData();
+  const { groups = [], dagRuns = [] } = data;
+  const durations = [];
+  const runs = dagRuns.map((dagRun) => {
+    const duration = getDuration(dagRun.startDate, dagRun.endDate);
+    durations.push(duration);
+    return {
+      ...dagRun,
+      duration,
+    };
+  });
+
+  // calculate dag run bar heights relative to max
+  const max = Math.max.apply(null, durations);
+
+  const renderGroup = (prevGroup, level) => prevGroup.children.map((group) => (
+    <TaskGroup key={group.id} group={group} level={level} prevGroup={prevGroup.id} />
+  ));
+
+  const TaskGroup = ({ prevGroup, group, level }) => {
+    const isGroup = !!group.children;
+
+    const storageKey = `${dagId}-open-groups`;
+    const openGroups = JSON.parse(localStorage.getItem(storageKey)) || [];
+    const isGroupId = openGroups.some((g) => g === group.id);
+
+    const onOpen = () => {
+      localStorage.setItem(storageKey, JSON.stringify([...openGroups, group.id]));
+    };
+    const onClose = () => {
+      localStorage.setItem(storageKey, JSON.stringify(openGroups.filter((g) => g !== group.id)));
+    };
+
+    const { isOpen, onToggle } = useDisclosure({ defaultIsOpen: isGroupId, onClose, onOpen });
+    const [isHover, setHover] = useState(false);
+    // Show only the immediate task id
+    const taskName = group.id.replace(`${prevGroup}.`, '');
+
+    return (
+      <Box backgroundColor={`rgba(203, 213, 224, ${0.25 * level})`}>
+        <Flex
+          justifyContent="space-between"
+          alignItems="center"
+          width="100%"
+          py="1px"
+          borderBottomWidth={1}
+          borderBottomColor={level > 1 ? 'white' : 'gray.200'}
+          backgroundColor={isHover && 'rgba(113, 128, 150, 0.1)'}
+          onMouseEnter={() => setHover(true)}
+          onMouseLeave={() => setHover(false)}
+        >
+          <Flex
+            as={isGroup ? 'button' : 'div'}
+            onClick={() => isGroup && onToggle()}
+            color={(level > 4 || (isHover && level > 3)) && 'white'}
```

Comment:
Wondering if you could avoid the `isHover` state with something like this?
```suggestion
          role="group"
          _hover={{ backgroundColor: 'rgba(113, 128, 150, 0.1'}}
        >
          <Flex
            as={isGroup ? 'button' : 'div'}
            onClick={() => isGroup && onToggle()}
            color={level > 4 ? 'white' : undefined}
            _groupHover={level > 3 && {
              color: 'white',
            }}
```